scalafix-core/shared/src/main/scala/scalafix/syntax/package.scala

```diff
@@ -1,24 +1,19 @@
 package scalafix
 
-import java.nio.charset.Charset
 import scala.collection.immutable.Seq
 import scala.meta._
-import scala.meta.semanticdb.Signature
 import scala.meta.semanticdb.Symbol
-import scala.util.Try
-import org.scalameta.logger
 import scala.compat.Platform.EOL
 import scala.meta.internal.scalafix.ScalafixScalametaHacks
 import scalafix.internal.util.SymbolOps
 import scalafix.util.TreeOps
 
 package object syntax {
 
-  implicit class XtensionRefSymbolOpt(ref: Ref)(implicit mirror: SemanticCtx) {
-    def symbolOpt: Option[Symbol] = mirror.names.collectFirst {
-      case ResolvedName(pos, sym, _) if pos == ref.pos => sym
-    }
-    def symbol: Symbol = symbolOpt.get
+  implicit class XtensionRefSymbolOpt(tree: Tree)(implicit mirror: SemanticCtx) {
+    @deprecated("Renamed to symbol", "0.5.0")
+    def symbolOpt: Option[Symbol] = symbol
```

Comment:
The only reason `.symbol` didn't return `Option[Symbol]` is because that's how it was in the scala.meta api. Now those methods are gone so we can do as we please 😄